Tools/peg_generator/pegen/c_generator.py

```diff
@@ -685,6 +681,12 @@ def handle_alt_loop(self, node: Alt, is_gather: bool, rulename: Optional[str]) -
     def visit_Alt(
         self, node: Alt, is_loop: bool, is_gather: bool, rulename: Optional[str]
     ) -> None:
+        self.print("if (p->error_indicator == 1 || PyErr_Occurred()) {")
+        with self.indent():
+            self.print("p->error_indicator = 1;")
+            self.print("return NULL;")
+        self.print("}")
```

Comment:
I'd actually move this within the scope block of the alternative (after the `with self.indent()` below).